app/server/lib/updateChecker.ts

```diff
@@ -0,0 +1,55 @@
+import {Response as ExpressRequest} from 'express';
+import {commonUrls} from "app/common/gristUrls";
+import {version as installedVersion, LatestVersionAvailable} from "app/common/version";
+import {naturalCompare} from 'app/common/SortFunc';
+import {GristServer} from "app/server/lib/GristServer";
+
+export async function checkForUpdates(gristServer: GristServer, res: ExpressRequest|null) {
```

Comment:
The type argument is wrong I think. It should be ExpressRequest.

Reply:
I've reworked this function to be closer to what I think Spoffy suggested. It's a small rewrite instead of just copying the code over, but I think it makes sense.